command/agent/job_endpoint.go

```diff
@@ -2060,3 +2060,48 @@ func validateEvalPriorityOpt(priority int) HTTPCodedError {
 	}
 	return nil
 }
+
+func (s *HTTPServer) JobsStatusesRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	args := structs.JobsStatusesRequest{}
+	if s.parse(resp, req, &args.Region, &args.QueryOptions) {
+		return nil, nil // seems whack
+	}
+	if smartOnly, err := parseBool(req, "smart_only"); err != nil {
+		return nil, err
+	} else if smartOnly != nil {
+		args.SmartOnly = *smartOnly
+	}
+	switch req.Method {
+	case http.MethodGet:
+		// GET requests will be treated as "get all jobs" but also with filtering and pagination and such
+	case http.MethodPost:
+		// POST requests expect a list of Jobs in the request body, which will then be filtered/paginated, etc.
+		var in api.JobsStatusesRequest
+		if err := decodeBody(req, &in); err != nil {
+			return nil, err
+		}
+		if len(in.Jobs) == 0 {
+			return nil, CodedError(http.StatusBadRequest, "no jobs in request")
+		}
```

Comment:
Likewise, here we're returning an error when we _could_ just do what the GET does and return all the things. Making the user switch between the two could be a little annoying. But maybe it doesn't matter if both the UI and Go API "does the right thing" and picks the correct method automatically.

Reply:
my instinct is to enshrine the spec, but I'll play around with this a bit and see how it feels.

Reply:
in [924935a](https://github.com/hashicorp/nomad/pull/20130/commits/924935a54f0882fd1fe92dddf5481d2bac5943a6) I treat GET and POST the same, but I still want to error if they do give a request body, but it is malformed.